From b45a19206751cd06973a7f00ab560cabc4b1626f Mon Sep 17 00:00:00 2001
From: Austin Shafer <ashafer@nvidia.com>
Date: Mon, 17 Oct 2022 16:15:46 -0400
Subject: [PATCH 2/3] conftest: Add old test_kernel_headers back

This was redone to be parallelized in the linux build, but this
is easier to do the original way so just add the original test
back in.
---
 .../arch/nvalloc/unix/Linux/conftest.sh       | 100 ++++++++++++++++++
 1 file changed, 100 insertions(+)

diff --git a/nvidia/src/nvidia-drm/conftest.sh b/nvidia/src/nvidia-drm/conftest.sh
index ccd1b6c2333..d2e4ba93790 100755
--- a/nvidia/src/nvidia-drm/conftest.sh
+++ b/nvidia/src/nvidia-drm/conftest.sh
@@ -7024,6 +7024,106 @@ case "$5" in
         exit 1
     ;;
 
+    test_kernel_headers)
+        #
+        # Determine which header files (of a set that may or may not be
+        # present) are provided by the target kernel.
+        #
+        FILES="asm/system.h"
+        FILES="$FILES drm/drmP.h"
+        FILES="$FILES drm/drm_auth.h"
+        FILES="$FILES drm/drm_gem.h"
+        FILES="$FILES drm/drm_crtc.h"
+        FILES="$FILES drm/drm_atomic.h"
+        FILES="$FILES drm/drm_atomic_helper.h"
+        FILES="$FILES drm/drm_encoder.h"
+        FILES="$FILES drm/drm_atomic_uapi.h"
+        FILES="$FILES drm/drm_drv.h"
+        FILES="$FILES drm/drm_framebuffer.h"
+        FILES="$FILES drm/drm_connector.h"
+        FILES="$FILES drm/drm_probe_helper.h"
+        FILES="$FILES drm/drm_blend.h"
+        FILES="$FILES drm/drm_fourcc.h"
+        FILES="$FILES drm/drm_prime.h"
+        FILES="$FILES drm/drm_plane.h"
+        FILES="$FILES drm/drm_vblank.h"
+        FILES="$FILES drm/drm_file.h"
+        FILES="$FILES drm/drm_ioctl.h"
+        FILES="$FILES drm/drm_device.h"
+        FILES="$FILES drm/drm_mode_config.h"
+        FILES="$FILES drm/drm_modeset_lock.h"
+        FILES="$FILES dt-bindings/interconnect/tegra_icc_id.h"
+        FILES="$FILES generated/autoconf.h"
+        FILES="$FILES generated/compile.h"
+        FILES="$FILES generated/utsrelease.h"
+        FILES="$FILES linux/efi.h"
+        FILES="$FILES linux/kconfig.h"
+        FILES="$FILES linux/platform/tegra/mc_utils.h"
+        FILES="$FILES linux/semaphore.h"
+        FILES="$FILES linux/printk.h"
+        FILES="$FILES linux/ratelimit.h"
+        FILES="$FILES linux/prio_tree.h"
+        FILES="$FILES linux/log2.h"
+        FILES="$FILES linux/of.h"
+        FILES="$FILES linux/bug.h"
+        FILES="$FILES linux/sched.h"
+        FILES="$FILES linux/sched/mm.h"
+        FILES="$FILES linux/sched/signal.h"
+        FILES="$FILES linux/sched/task.h"
+        FILES="$FILES linux/sched/task_stack.h"
+        FILES="$FILES xen/ioemu.h"
+        FILES="$FILES linux/fence.h"
+        FILES="$FILES linux/dma-resv.h"
+        FILES="$FILES soc/tegra/chip-id.h"
+        FILES="$FILES soc/tegra/fuse.h"
+        FILES="$FILES soc/tegra/tegra_bpmp.h"
+        FILES="$FILES video/nv_internal.h"
+        FILES="$FILES linux/platform/tegra/dce/dce-client-ipc.h"
+        FILES="$FILES linux/nvhost.h"
+        FILES="$FILES linux/nvhost_t194.h"
+        FILES="$FILES asm/book3s/64/hash-64k.h"
+        FILES="$FILES asm/set_memory.h"
+        FILES="$FILES asm/prom.h"
+        FILES="$FILES asm/powernv.h"
+        FILES="$FILES linux/atomic.h"
+        FILES="$FILES asm/barrier.h"
+        FILES="$FILES asm/opal-api.h"
+        FILES="$FILES sound/hdaudio.h"
+        FILES="$FILES asm/pgtable_types.h"
+        FILES="$FILES linux/stringhash.h"
+        FILES="$FILES linux/dma-map-ops.h"
+        FILES="$FILES rdma/peer_mem.h"
+        FILES="$FILES sound/hda_codec.h"
+        FILES="$FILES linux/dma-buf.h"
+        FILES="$FILES linux/time.h"
+        FILES="$FILES linux/platform_device.h"
+        FILES="$FILES linux/mutex.h"
+        FILES="$FILES linux/reset.h"
+        FILES="$FILES linux/of_platform.h"
+        FILES="$FILES linux/of_device.h"
+        FILES="$FILES linux/of_gpio.h"
+        FILES="$FILES linux/gpio.h"
+        FILES="$FILES linux/gpio/consumer.h"
+        FILES="$FILES linux/interconnect.h"
+        FILES="$FILES linux/pm_runtime.h"
+        FILES="$FILES linux/clk.h"
+        FILES="$FILES linux/clk-provider.h"
+        FILES="$FILES linux/ioasid.h"
+        FILES="$FILES linux/stdarg.h"
+        FILES="$FILES linux/iosys-map.h"
+        FILES="$FILES asm/coco.h"
+        FILES="$FILES linux/vfio_pci_core.h"
+        FILES="$FILES soc/tegra/bpmp-abi.h"
+        FILES="$FILES soc/tegra/bpmp.h"
+
+        CFLAGS=$6
+
+        for file in $FILES ; do
+            test_header_presence "$file"
+        done
+
+        exit 0
+    ;;
 
     test_kernel_header)
         #
-- 
2.41.0

